internal/provider/resources/resource_api_token_test.go

```diff
@@ -0,0 +1,568 @@
+					resource.TestCheckResourceAttr(resourceVar, "roles.1.role", "WORKSPACE_OWNER"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.2.entity_id", deploymentId),
+					resource.TestCheckResourceAttr(resourceVar, "roles.2.entity_type", "DEPLOYMENT"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.2.role", "DEPLOYMENT_ADMIN"),
+					// Check via API that organization api token exists
+					testAccCheckApiTokenExistence(t, checkApiTokensExistenceInput{name: apiTokenName, organization: true, shouldExist: true}),
+				),
+			},
+			// Change properties and check they have been updated in terraform state
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: "new description",
+					Type:        "ORGANIZATION",
+					Roles: []apiTokenRole{
+						{
+							Role:       "ORGANIZATION_OWNER",
+							EntityId:   organizationId,
+							EntityType: "ORGANIZATION",
+						},
+						{
+							Role:       "WORKSPACE_OWNER",
+							EntityId:   workspaceId,
+							EntityType: "WORKSPACE",
+						},
+						{
+							Role:       "DEPLOYMENT_ADMIN",
+							EntityId:   deploymentId,
+							EntityType: "DEPLOYMENT",
+						},
+					},
+					ExpiryPeriodInDays: 30,
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceVar, "description", "new description"),
+					// Check via API that organization api token exists
+					testAccCheckApiTokenExistence(t, checkApiTokensExistenceInput{name: apiTokenName, organization: true, shouldExist: true}),
+				),
+			},
+			// Change the resource type and remove roles
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: utils.TestResourceDescription,
+					Type:        "WORKSPACE",
+					Roles: []apiTokenRole{
+						{
+							Role:       "WORKSPACE_OWNER",
+							EntityId:   workspaceId,
+							EntityType: "WORKSPACE",
+						},
+					},
+					ExpiryPeriodInDays: 30,
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceVar, "type", "WORKSPACE"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.#", "1"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.0.entity_id", workspaceId),
+					resource.TestCheckResourceAttr(resourceVar, "roles.0.entity_type", "WORKSPACE"),
+					resource.TestCheckResourceAttr(resourceVar, "roles.0.role", "WORKSPACE_OWNER"),
+					// Check via API that api token was destroyed and recreated
+					testAccCheckApiTokenExistence(t, checkApiTokensExistenceInput{name: apiTokenName, workspace: true, shouldExist: true}),
+				),
+			},
+			// Change resource type back to ORGANIZATION
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiToken(apiTokenInput{
+					Name:        apiTokenName,
+					Description: utils.TestResourceDescription,
```

Comment:
lets remove the optional fields for this test and add that to the comment above and see if it works
```suggestion
```